hypertrace-graphql-gateway-service-metric-utils/src/main/java/org/hypertrace/graphql/utils/metrics/gateway/BaselineIntervalListContainer.java

```diff
@@ -0,0 +1,180 @@
+package org.hypertrace.graphql.utils.metrics.gateway;
+
+import io.reactivex.rxjava3.core.Observable;
+import io.reactivex.rxjava3.core.Single;
+import io.reactivex.rxjava3.observables.GroupedObservable;
+import lombok.Getter;
+import lombok.Value;
+import lombok.experimental.Accessors;
+import org.hypertrace.core.graphql.common.utils.CollectorUtils;
+import org.hypertrace.core.graphql.common.utils.Converter;
+import org.hypertrace.gateway.service.v1.baseline.Baseline;
+import org.hypertrace.gateway.service.v1.baseline.BaselineInterval;
+import org.hypertrace.gateway.service.v1.baseline.BaselineMetricSeries;
+import org.hypertrace.graphql.metric.request.MetricAggregationRequest;
+import org.hypertrace.graphql.metric.request.MetricSeriesRequest;
+import org.hypertrace.graphql.metric.schema.BaselineMetricInterval;
+import org.hypertrace.graphql.metric.schema.BaselinedMetricAggregation;
+import org.hypertrace.graphql.metric.schema.Health;
+import org.hypertrace.graphql.metric.schema.MetricBaselineAggregation;
+
+import javax.inject.Inject;
+import java.time.Instant;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+class BaselineIntervalListContainer
```

Comment:
make sure all the class names match the classes they produce - here BaselineMetricInterval rather than BaselineInterval